src/extension/ddappsec.c

```diff
@@ -225,22 +227,30 @@ static PHP_MSHUTDOWN_FUNCTION(ddappsec)
     return SUCCESS;
 }
 
-static pthread_once_t _rinit_config_once_control = PTHREAD_ONCE_INIT;
+static pthread_once_t _rinit_once_control = PTHREAD_ONCE_INIT;
 
 static PHP_RINIT_FUNCTION(ddappsec)
 {
-    pthread_once(&_rinit_config_once_control, dd_config_first_rinit);
+    pthread_once(&_rinit_once_control, dd_config_first_rinit);
     zai_config_rinit();
 
-    _check_enabled();
+    //_check_enabled should be run only once. However, pthread_once approach
+    // does not work with ZTS.
+    if (DDAPPSEC_G(enabled) == NOT_CONFIGURED) {
+        mlog_g(dd_log_trace,
+            "[Remote config] Enabled not configured, computing enabled status");
+        _check_enabled();
+    }
+    mlog_g(dd_log_trace, "[Remote config] Enable status is %d",
```

Comment:
Don't add `[Remote config]` to the logs, also I don't think you need this log statement anymore.

Reply:
Done